app/client/cli/cache/session.go

```diff
@@ -0,0 +1,87 @@
+package cache
+
+// TODO: add a TTL for cached sessions, since we know the sessions' length
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/pokt-network/pocket/persistence/kvstore"
+	"github.com/pokt-network/pocket/rpc"
+)
+
+var errSessionNotFound = errors.New("session not found in cache")
+
+// SessionCache defines the set of methods used to interact with the client-side session cache
+type SessionCache interface {
+	Get(appAddr, chain string) (*rpc.Session, error)
+	Set(session *rpc.Session) error
+	Stop() error
+}
+
+// sessionCache stores and retrieves sessions for application+relaychain pairs
+//
+//	It uses a key-value store as backing storage
+type sessionCache struct {
+	// store is the local store for cached sessions
+	store kvstore.KVStore
+}
+
+// Create returns a session cache backed by a kvstore using the provided database path.
+func Create(databasePath string) (SessionCache, error) {
```

Comment:
```suggestion
func NewSessionCache(databasePath string) (SessionCache, error) {
```

Personally I would change this to avoid any confusion with this being a submodule. + `NewSessionCache` is far more descriptive

Reply:
Agreed with @h5law on this

Reply:
Thank you for the reviews. Fixed.